pkg/jsonmessage/jsonmessage.go

```diff
@@ -189,13 +189,9 @@ func DisplayJSONMessagesStream(in io.Reader, out io.Writer, terminalFd uintptr,
 				if isTerminal {
 					fmt.Fprintf(out, "\n")
 				}
-			} else {
-				diff = len(ids) - line
 			}
-			if isTerminal {
-				// NOTE: this appears to be necessary even if
-				// diff == 0.
-				// <ESC>[{diff}A = move cursor up diff rows
+			diff = len(ids) - line
```

Comment:
Oh wait, I misread your comment, sorry. You are talking about `line` not `diff`. You are correct that this is undefined in the !ok case. I think this needs an actual assignment and not an assumption/comment that line is zero. I shall fix that.